### DISCUSSION TITLE
Seeking any scripts to obtain numbers from a map.dx file

### DISCUSSION THREAD

**Discussion by limeiling-r** [Electrostatics/apbs#170] [Q&A]
Hello,

I calculated an ion channel protein’s electrostatic potential by APBS web version and generated a map.dx file, which contains protein electrostatic potential numbers. 

Would you please provide me with any scripts or data processing software that can help me obtain the electrostatic potential along the pore of an ion channel?

The start and end of the step2_pbeq.dx file as follows:

#  
object 1 class gridpositions counts 145 139 191
origin 82.614059 83.660582 65.927911
delta 1.000000 0.0 0.0
delta 0.0 1.000000 0.0
delta 0.0 0.0 1.000000
object 2 class gridconnections counts 145 139 191
object 3 class array type double rank 0 items 3849605 data follows
-0.000019 -0.000013 -0.000003
0.000016 0.000028 0.000033
0.000045 0.000053 0.000057
…
…
…
0.000066 0.000058 0.000051
0.000047 0.000044 0.000043
0.000000 0.000000 0.000000
attribute "dep" string "positions"
object "irregular positions irregular connections" class field
component "positions" value 1
component "connections" value 2
component "data" value 3
end

Thanks for your reading and hoping to hear from you.

Best Wishes!

**Comment by sobolevnrm** [comment]
The APBS `multivalue` tool should provide potential values at user-specified locations.

**Comment by intendo** [comment]
The multivalue source is in `$APBS_SRC_DIR/tools/mesh/multivalue.c` and the binary gets installed in `$INSTALL_DIR/apbs/share/apbs/tools/bin/multivalue`
